internal/pkg/resource/resource.go

```diff
@@ -70,3 +93,88 @@ func LookupType(resourceId string) string {
 
 	return Unknown
 }
+
+func ValidatePrefixes(resourceType string, args []string) error {
+	prefix, ok := resourceToPrefix[resourceType]
+	if !ok {
+		return nil
+	}
+
+	var malformed []string
+	for _, resourceId := range args {
+		if LookupType(resourceId) != resourceType {
+			malformed = append(malformed, resourceId)
+		}
+	}
+
+	if len(malformed) == 1 {
+		return errors.Errorf(`failed parsing resource ID %s: missing prefix "%s-"`, malformed[0], prefix)
+	} else if len(malformed) > 1 {
+		return errors.Errorf(`failed parsing resource IDs %s: missing prefix "%s-"`, utils.ArrayToCommaDelimitedString(malformed, "and"), prefix)
+	}
+
+	return nil
+}
+
+func ValidateArgs(fullParentCommand string, args []string, resourceType string, callDescribeEndpoint func(string) error) error {
+	var invalidArgs []string
+	for _, arg := range args {
+		if err := callDescribeEndpoint(arg); err != nil {
+			invalidArgs = append(invalidArgs, arg)
+		}
+	}
+
+	if len(invalidArgs) != 0 {
+		NotFoundErrorMsg := `%s %s not found`
+		invalidArgsErrMsg := fmt.Sprintf(NotFoundErrorMsg, resourceType, utils.ArrayToCommaDelimitedString(invalidArgs, "and"))
+		if len(invalidArgs) > 1 {
+			invalidArgsErrMsg = fmt.Sprintf(NotFoundErrorMsg, Plural(resourceType), utils.ArrayToCommaDelimitedString(invalidArgs, "and"))
+		}
+		return errors.NewErrorWithSuggestions(invalidArgsErrMsg, fmt.Sprintf(errors.ListResourceSuggestions, resourceType, fullParentCommand))
+	}
+
+	return nil
+}
+
+func Plural(resource string) string {
+	if resource == "" {
+		return ""
+	}
+
+	for suffix := range pluralExtraSyllableSuffix {
+		if strings.HasSuffix(resource, suffix) {
+			return resource + "es"
+		}
+	}
+
+	return resource + "s"
+}
+
+func DefaultPostProcess(_ string) error {
```

Comment:
Can we just pass `nil` instead of this function? It's unclear at first glance what `DefaultPostProcess` means/does.

Reply:
Oh, good point.